lib/src/main/java/com/squareup/cycler/Update.kt

```diff
@@ -120,49 +166,68 @@ class Update<I : Any>(private val oldRecyclerData: RecyclerData<I>) {
   internal fun generateUpdateWork(itemComparator: ItemComparator<I>?): UpdateWork {
 
     val extraItemChanged = oldRecyclerData.extraItem != extraItem
-    val refreshAllNeeded = dataReplaced && itemComparator == null
 
     val asyncWork = mutableListOf<() -> Unit>()
     val notifications = mutableListOf<(Adapter<*>) -> Unit>()
 
-    if (refreshAllNeeded) {
-      @SuppressLint("NotifyDataSetChanged")
-      notifications += { adapter -> adapter.notifyDataSetChanged() }
-    } else {
+    if (extraItemChanged) {
+      // We notify first the extraItem change, so we use its position in oldRecyclerData.
+      notifications += notifyChangesExtraItem()
+    }
```

Comment:
This should be done only if we are not replacing everything (we will know that inside this method itself).

Reply:
In other words: if !dataPopulated (either updated or added we update extra item independently)